src/flamenco/runtime/program/fd_system_program_nonce.c

```diff
@@ -961,51 +961,97 @@ fd_load_nonce_account( fd_exec_txn_ctx_t const *   txn_ctx,
   return 1;
 }
 
-
+/* https://github.com/firedancer-io/solana/blob/4b31032e68f85848b02fcc4c9e580d57f32ec04b/runtime/src/bank.rs#L4755-L4756 */
 int
-fd_has_nonce_account( fd_exec_txn_ctx_t const *   txn_ctx,
-                       int *                      perr ) {
+fd_has_nonce_account( fd_exec_txn_ctx_t const * txn_ctx, int * perr ) {
+  ushort recent_blockhash_off = txn_ctx->txn_descriptor->recent_blockhash_off;
+  fd_hash_t * recent_blockhash = (fd_hash_t *)((uchar *)txn_ctx->_txn_raw->raw + recent_blockhash_off);
 
-  *perr = 0;
+  fd_block_block_hash_entry_t * hashes_deque   = txn_ctx->slot_ctx->slot_bank.recent_block_hashes.hashes;
```

Comment:
I think this comes from the block hash queue: https://github.com/firedancer-io/solana/blob/4b31032e68f85848b02fcc4c9e580d57f32ec04b/runtime/src/bank.rs#L4654

Use slot_ctx->slot_bank.block_hash_queue